src/components/Canvas.tsx

```diff
@@ -133,12 +131,11 @@ const Canvas: React.FC<CanvasProps> = ({ data, selectedBits, isGridView }) => {
   ]);
 
   const processBatch = useCallback(() => {
-    // Process the batch data
-    if (batchBuffer.length === 0) return;
+    if (batchBuffer.length === 0 || !isDisplay) return;
 
     batchBuffer.forEach((batch) => {
       channels.forEach((channel, index) => {
-        if (channel && !isPaused[index]) {
+        if (channel) {
```

Comment:
Can you explain reason to remove !isPaused[index] from condition check? It was making sure that data won't append to a stream which is paused.